src/main/java/org/occiware/tinom/aggregators/OrderedWeightedAveraging.java

```diff
@@ -0,0 +1,177 @@
+        }
+    };
+
+    public static final ParametricFactory<Aggregation> MEOWA = new ParametricFactory<Aggregation>() {
+        @Override
+        public Aggregation create(double... params) {
+            if (params == null || params.length < 1)
+                throw new IllegalArgumentException("MEOWA factory requires one parameter (rho)");
+            final double rho = params[0];
+            // aggr
+            return new Aggregation() {
+                private OrderedWeightedAveraging owa;
+                @Override
+                public double apply(double... values) {
+                    if (owa == null)
+                        owa = MEOWA_FACTORY(values.length, rho);
+                    return owa.apply(values);
+                }
+            };
+        }
+    };
+
+    /**
+     * Construct OWA operator with MEOWA weights.
+     *
+     * @param n Number of weights
+     * @param rho The andness
+     * @return OWA with weights calculated
+     */
+    public static OrderedWeightedAveraging MEOWA_FACTORY(int n, double rho) {
+
+        if (rho < 0.0 || rho > 1.0)
+            throw new IllegalArgumentException("andness (rho) must be within [0, 1]");
+
+        if (rho == 1.0) { // handle border-case, rho = 1.0
+            double[] w = new double[n];
+            Arrays.fill(w, 0.0);
+            w[n-1] = 1.0;
+            return new OrderedWeightedAveraging(w);
+        }
+
+
+        // calculate the two roots for rho
+        double t_m = (-(rho - 0.5) - Math.sqrt(((rho - 0.5) * (rho - 0.5)) - (4 * (rho - 1) * rho))) / (2 * (rho - 1));
+        double t_p = (-(rho - 0.5) + Math.sqrt(((rho - 0.5) * (rho - 0.5)) - (4 * (rho - 1) * rho))) / (2 * (rho - 1));
+        double t = Math.max(t_m, t_p);
+
+        // System.out.println("t = " + t);
+
+        double[] w = new double[n];
+        double s = 0.0;
+        for (int i = 0; i < n; i++) {
+            w[i] = Math.pow(t, i);
+            s += w[i];
+        }
+        for (int i = 0; i < n; i++) {
+            w[i] /= s;
+        }
+
+        return new OrderedWeightedAveraging(w);
+    }
+
+
+    public final double[] weights;
+
+    public OrderedWeightedAveraging(double... weights) {
+        this.weights = weights;
+        double sum = 0.0;
+        for (double w : weights)
+            sum += w;
+        double diff = Math.abs(1.0 - sum);
+        if (diff > 0.0001)
+            throw new IllegalArgumentException(String.format("sum(weights) must be 1.0 (was: %f, %.5f)", sum, diff));
+    }
+
+    @Override
+    public double apply(double... values) {
+        // we need same lenghts
+        int min = Math.min(values.length, weights.length);
+        // sort
+        Arrays.sort(values);
+        OrderedWeightedAveraging.reverse(values);
+        //
+        double s = 0.0;
+        for (int i = 0; i < min; i++)
+            s += values[i] * weights[i];
+
+        return s;
+    }
+
+    /**
+     * Calculate the orness of this OWA operator.
```

Comment:
According to Sun's guidelines for Javadoc writing, you should write **Calculates** and not **Calculate**.
I know Sun is dead, but their guidelines are still a reference. :wink: